datasets/ade_corpus_v2/ade_corpus_v2.py

```diff
@@ -0,0 +1,272 @@
+
+import datasets
+
+
+_CITATION = """\
+@article{GURULINGAPPA2012885,
+title = "Development of a benchmark corpus to support the automatic extraction of drug-related adverse effects from medical case reports",
+journal = "Journal of Biomedical Informatics",
+volume = "45",
+number = "5",
+pages = "885 - 892",
+year = "2012",
+note = "Text Mining and Natural Language Processing in Pharmacogenomics",
+issn = "1532-0464",
+doi = "https://doi.org/10.1016/j.jbi.2012.04.008",
+url = "http://www.sciencedirect.com/science/article/pii/S1532046412000615",
+author = "Harsha Gurulingappa and Abdul Mateen Rajput and Angus Roberts and Juliane Fluck and Martin Hofmann-Apitius and Luca Toldo",
+keywords = "Adverse drug effect, Benchmark corpus, Annotation, Harmonization, Sentence classification",
+abstract = "A significant amount of information about drug-related safety issues such as adverse effects are published in medical case reports that can only be explored by human readers due to their unstructured nature. The work presented here aims at generating a systematically annotated corpus that can support the development and validation of methods for the automatic extraction of drug-related adverse effects from medical case reports. The documents are systematically double annotated in various rounds to ensure consistent annotations. The annotated documents are finally harmonized to generate representative consensus annotations. In order to demonstrate an example use case scenario, the corpus was employed to train and validate models for the classification of informative against the non-informative sentences. A Maximum Entropy classifier trained with simple features and evaluated by 10-fold cross-validation resulted in the F1 score of 0.70 indicating a potential useful application of the corpus."
+}
+"""
+
+_DESCRIPTION = """\
+ ADE-Corpus-V2  Dataset: Adverse Drug Reaction Data.
+ This is a dataset for Classification if a sentence is ADE-related (True) or not (False) and Relation Extraction between Adverse Drug Event and Drug.
+ DRUG-AE.rel provides relations between drugs and adverse effects.
+ DRUG-DOSE.rel provides relations between drugs and dosages.
+ ADE-NEG.txt provides all sentences in the ADE corpus that DO NOT contain any drug-related adverse effects.
+"""
+
+_DOWNLOAD_URL = "https://raw.githubusercontent.com/trunghlt/AdverseDrugReaction/master/ADE-Corpus-V2/{}-{}.{}"
+
+# Different usage configs/
+configs = {
+    "classifiaction": "Ade_corpos_v2_classificaion",
+    "RE_ade": "Ade_corpos_v2_drug_ade_relation",
+    "RE_dosage": "Ade_corpos_v2_drug_dosage_relation",
+}
+
+
+class ADE_Corpus_V2Config(datasets.BuilderConfig):
+    """BuilderConfig for ADE_Corpus_V2."""
+
+    def __init__(self, **kwargs):
+        """BuilderConfig for ADE_Corpus_V2.
+        Args:
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(ADE_Corpus_V2Config, self).__init__(**kwargs)
+
+
+class ADE_Corpus_V2(datasets.GeneratorBasedBuilder):
```

Comment:
```suggestion
class ADECorpusV2(datasets.GeneratorBasedBuilder):
```
this must be the camel case version of the dataset script name `ade_corpus_v2`